golint: Fix all the golint complaints

```
$ golint ./...
copy-object-if-modified-since.go:91:9: if block ends with a return statement, so drop this else and outdent its block
copy-object-if-none-match.go:88:9: if block ends with a return statement, so drop this else and outdent its block
copy-object-if-unmodified-since.go:91:9: if block ends with a return statement, so drop this else and outdent its block
data-types.go:57:1: comment on exported type ObjectInfos should be of the form "ObjectInfos ..." (with optional leading article)
data-types.go:96:1: comment on exported type ObjectMultipartInfo should be of the form "ObjectMultipartInfo ..." (with optional leading article)
globals.go:33:37: should drop = nil from declaration of var globalRandom; it is the zero value
main.go:59:1: comment on exported type APItest should be of the form "APItest ..." (with optional leading article)
prepare-s3verify.go:95:21: error strings should not end with punctuation
put-bucket.go:143:9: if block ends with a return statement, so drop this else and outdent its block
retry.go:37:5: exported var TCPretry should have comment or be unexported
server-config.go:28:6: exported type ServerConfig should have comment or be unexported
signv4/request-signature-v4.go:261:1: comment on exported function SignV4 should be of the form "SignV4 ..."
signv4/signv4-utils.go:37:1: comment on exported function Sum256 should be of the form "Sum256 ..."
```

To check for this run this automatically in '.travis.yml' - refer minio server travis.

--- a/README.md
+++ b/README.md
@@ -61,7 +61,7 @@ Use s3verify to check the AWS S3 V4 compatibility of the Minio test server (http
 $ s3verify -a YOUR_ACCESS_KEY -s YOUR_SECRET_KEY https://play.minio.io:9000 
 ```
 
-Use s3verify to check the AWS S3 V4 compatability of the Minio test server with all APIs.
+Use s3verify to check the AWS S3 V4 compatibility of the Minio test server with all APIs.
 ```
 $ s3verify -a YOUR_ACCESS_KEY -s YOUR_SECERT_KEY https://play.minio.io:9000 --extended
 ```

--- a/copy-object-if-modified-since.go
+++ b/copy-object-if-modified-since.go
@@ -88,19 +88,19 @@ func verifyBodyCopyObjectIfModifiedSince(resBody io.Reader, expectedError ErrorR
 			return err
 		}
 		return nil
-	} else {
-		copyObjResult := copyObjectResult{}
-		err := xmlDecoder(resBody, &copyObjResult)
-		if err != nil {
-			body, errR := ioutil.ReadAll(resBody)
-			if errR != nil {
-				return errR
-			}
-			err = fmt.Errorf("Unexpected Body Received: %v", string(body))
-			return err
-		}
-		return nil
 	}
+	// Verify the body returned is a copyObject result.
+	copyObjResult := copyObjectResult{}
+	err := xmlDecoder(resBody, &copyObjResult)
+	if err != nil {
+		body, errR := ioutil.ReadAll(resBody)
+		if errR != nil {
+			return errR
+		}
+		err = fmt.Errorf("Unexpected Body Received: %v", string(body))
+		return err
+	}
+	return nil
 }
 
 // verifyHeaderCopyObjectIfModifiedSince - verify the header returned matches what is expected.

--- a/copy-object-if-none-match.go
+++ b/copy-object-if-none-match.go
@@ -85,13 +85,13 @@ func verifyBodyCopyObjectIfNoneMatch(resBody io.Reader, expectedError ErrorRespo
 			return err
 		}
 		return nil
-	} else { // Successful copy expected.
-		copyObjRes := copyObjectResult{}
-		if err := xmlDecoder(resBody, &copyObjRes); err != nil {
-			return err
-		}
-		return nil
 	}
+	// Successful copy expected.
+	copyObjRes := copyObjectResult{}
+	if err := xmlDecoder(resBody, &copyObjRes); err != nil {
+		return err
+	}
+	return nil
 }
 
 // verifyHeaderCopyIfNoneMatch - Verify that the header returned matches what is expected.

--- a/copy-object-if-unmodified-since.go
+++ b/copy-object-if-unmodified-since.go
@@ -88,21 +88,20 @@ func verifyBodyCopyObjectIfUnModifiedSince(resBody io.Reader, expectedError Erro
 			return err
 		}
 		return nil
-	} else {
-		// Verify the body returned is a copyobject result.
-		copyObjResult := copyObjectResult{}
-		err := xmlDecoder(resBody, &copyObjResult)
-		if err != nil {
-			body, errR := ioutil.ReadAll(resBody)
-			if errR != nil {
-				return errR
-			}
-			err = fmt.Errorf("Unexpected Body Received: %v", string(body))
-
-			return err
-		}
-		return nil
 	}
+	// Verify the body returned is a copyobject result.
+	copyObjResult := copyObjectResult{}
+	err := xmlDecoder(resBody, &copyObjResult)
+	if err != nil {
+		body, errR := ioutil.ReadAll(resBody)
+		if errR != nil {
+			return errR
+		}
+		err = fmt.Errorf("Unexpected Body Received: %v", string(body))
+
+		return err
+	}
+	return nil
 }
 
 // verifyHeaderCopyObjectIfUnModifiedSince - verify the header returned matches what is expected.

--- a/data-types.go
+++ b/data-types.go
@@ -54,7 +54,7 @@ type ObjectInfo struct {
 	UploadID string // To be set only for multipart uploaded objects.
 }
 
-// A container for ObjectInfo structs to allow sorting.
+// ObjectInfos - A container for ObjectInfo structs to allow sorting.
 type ObjectInfos []ObjectInfo
 
 // Return the len of a list of ObjectInfo.
@@ -72,28 +72,7 @@ func (o ObjectInfos) Swap(i, j int) {
 	o[i], o[j] = o[j], o[i]
 }
 
-// objectInfoChannel a channel for concurrent object level operations.
-type objectInfoChannel struct {
-	objInfo ObjectInfo
-	err     error
-	index   int
-}
-
-// partChannel a channel for concurrent multipart upload part operations.
-type partChannel struct {
-	objPart objectPart
-	err     error
-	index   int
-}
-
-// multiUploadInitChannel a channel for concurrent multipart initiate upload operations.
-type multiUploadInitChannel struct {
-	uploadID string
-	err      error
-	index    int
-}
-
-// Container for a Multipart Objects information.
+// ObjectMultipartInfo - Container for a Multipart Objects information.
 type ObjectMultipartInfo struct {
 	// Date and time at which the multipart upload was initiated.
 	Initiated time.Time `type:"timestamp" timestampFormat:"iso8601"`

--- a/error-response.go
+++ b/error-response.go
@@ -155,21 +155,3 @@ func ErrInvalidArgument(message string) error {
 		RequestID: "minio",
 	}
 }
-
-// ErrInvalidBucketName - Invalid bucket name response.
-func ErrInvalidBucketName(message string) error {
-	return ErrorResponse{
-		Code:      "InvalidBucketName",
-		Message:   message,
-		RequestID: "minio",
-	}
-}
-
-// ErrInvalidObjectName - Invalid object name response.
-func ErrInvalidObjectName(message string) error {
-	return ErrorResponse{
-		Code:      "NoSuchKey",
-		Message:   message,
-		RequestID: "minio",
-	}
-}

--- a/get-object-if-none-match.go
+++ b/get-object-if-none-match.go
@@ -87,7 +87,7 @@ func verifyBodyGetObjectIfNoneMatch(resBody io.Reader, expectedBody []byte) erro
 	if err != nil {
 		return err
 	}
-	if !bytes.Equal(body, expectedBody) { // If the request does not go through an empty body is recieved.
+	if !bytes.Equal(body, expectedBody) { // If the request does not go through an empty body is received.
 		err := fmt.Errorf("Unexpected Body Recieved: wanted %v, got %v", string(expectedBody), string(body))
 		return err
 	}

--- a/globals.go
+++ b/globals.go
@@ -26,11 +26,10 @@ import (
 )
 
 var (
-	globalVerbose                    = false
-	globalDefaultRegion              = "us-east-1"
-	globalTotalNumTest               = 0
-	globalRequestPoolSize            = 7 // The maximum number of requests to be sent at once.
-	globalRandom          *rand.Rand = nil
+	globalVerbose       = false
+	globalDefaultRegion = "us-east-1"
+	globalTotalNumTest  = 0
+	globalRandom        *rand.Rand
 )
 
 // lockedRandSource provides protected rand source, implements rand.Source interface.

--- a/head-object.go
+++ b/head-object.go
@@ -74,7 +74,7 @@ func verifyStatusHeadObject(respStatusCode, expectedStatusCode int) error {
 	return nil
 }
 
-// verifyBodyHeadObject - Verify that the body recieved is empty.
+// verifyBodyHeadObject - Verify that the body received is empty.
 func verifyBodyHeadObject(resBody io.Reader) error {
 	body, err := ioutil.ReadAll(resBody)
 	if err != nil {

--- a/main.go
+++ b/main.go
@@ -56,7 +56,7 @@ EXAMPLES:
 		$ s3verify --access YOUR_ACCESS_KEY --secret YOUR_SECRET_KEY --url https://s3.amazonaws.com --region us-west-1
 `
 
-// Define all mainXXX tests to be of this form.
+// APItest - Define all mainXXX tests to be of this form.
 type APItest struct {
 	Test     func(ServerConfig, int) bool
 	Extended bool // Extended tests will only be invoked at the users request.

--- a/prepare-s3verify.go
+++ b/prepare-s3verify.go
@@ -92,7 +92,7 @@ func validateBucket(config ServerConfig, bucketName string) error {
 	// Validate the buckets name as being created by s3verify.
 	bucketNameParts := strings.Split(bucketName, "-")
 	if bucketNameParts[0] != "s3verify" {
-		err := fmt.Errorf("%s is not an s3verify created bucket. See s3verify --help.", bucketName)
+		err := fmt.Errorf("%s is not an s3verify created bucket. See s3verify --help", bucketName)
 		return err
 	}
 	validBucket := BucketInfo{

--- a/put-bucket.go
+++ b/put-bucket.go
@@ -140,16 +140,16 @@ func verifyBodyPutBucket(resBody io.Reader, expectedError ErrorResponse) error {
 			return err
 		}
 		return nil
-	} else {
-		body, err := ioutil.ReadAll(resBody)
-		if err != nil {
-			return err
-		}
-		// There is no body returned by a Put Bucket request.
-		if string(body) != "" {
-			err := fmt.Errorf("Unexpected Body: %v", string(body))
-			return err
-		}
+	}
+	// Read the body to make sure it is empty.
+	body, err := ioutil.ReadAll(resBody)
+	if err != nil {
+		return err
+	}
+	// There is no body returned by a Put Bucket request.
+	if string(body) != "" {
+		err := fmt.Errorf("Unexpected Body: %v", string(body))
+		return err
 	}
 	return nil
 }

--- a/put-object.go
+++ b/put-object.go
@@ -104,7 +104,7 @@ func verifyBodyPutObject(resBody io.Reader) error {
 	return nil
 }
 
-// verifyHeaderPutObject - Verify that the header returned matches waht is expected.
+// verifyHeaderPutObject - Verify that the header returned matches what is expected.
 func verifyHeaderPutObject(header http.Header) error {
 	if err := verifyStandardHeaders(header); err != nil {
 		return err

--- a/retry.go
+++ b/retry.go
@@ -34,6 +34,7 @@ const MaxJitter = 1.0
 // NoJitter disables the use of jitter for randomizing the exponential backoff time
 const NoJitter = 0.0
 
+// TCPretry holds all the errors that can and should be retried.
 var TCPretry = []string{"i/o timeout", "net/http: TLS handshake timeout", "connection reset by peer", "read: operation timed out"}
 
 // newRetryTimer creates a timer with exponentially increasing delays

--- a/scan-bar.go
+++ b/scan-bar.go
@@ -51,7 +51,6 @@ type scanBarFunc func(string)
 // scanBarFactory returns a progress bar function to report URL scanning.
 func scanBarFactory() scanBarFunc {
 	prevLineSize := 0
-	prevMessage := ""
 	termWidth, e := pb.GetTerminalWidth()
 	if e != nil {
 		console.Fatalln("Unable to get terminal size. Please use --quiet option.")
@@ -59,10 +58,6 @@ func scanBarFactory() scanBarFunc {
 
 	return func(message string) {
 		scanPrefix := fmt.Sprintf("%s", message)
-		eraseLen := prevLineSize
-		if eraseLen < 1 {
-			eraseLen = 0
-		}
 		padding := messageWidth - len([]rune(scanPrefix))
 
 		message = fixateScanBar(message, termWidth-len([]rune(scanPrefix))-1)
@@ -72,7 +67,6 @@ func scanBarFactory() scanBarFunc {
 			console.PrintC("\r")
 		}
 		console.PrintC(barText + " \b ") // Remove only the cursor from the text.
-		prevMessage = message
 		prevLineSize = len([]rune(barText))
 	}
 }

--- a/server-config.go
+++ b/server-config.go
@@ -25,6 +25,8 @@ import (
 	"github.com/minio/mc/pkg/httptracer"
 )
 
+// ServerConfig - container for all the user passed server info
+// and a reusable http.Client
 type ServerConfig struct {
 	Access   string
 	Secret   string

--- a/signv4/request-signature-v4.go
+++ b/signv4/request-signature-v4.go
@@ -200,7 +200,7 @@ func getCanonicalRequest(req http.Request) string {
 func getStringToSignV4(t time.Time, location, canonicalRequest string) string {
 	stringToSign := signV4Algorithm + "\n" + t.Format(iso8601DateFormat) + "\n"
 	stringToSign = stringToSign + getScope(location, t) + "\n"
-	stringToSign = stringToSign + hex.EncodeToString(Sum256([]byte(canonicalRequest)))
+	stringToSign = stringToSign + hex.EncodeToString(sum256([]byte(canonicalRequest)))
 	return stringToSign
 }
 
@@ -258,7 +258,7 @@ func postPresignSignatureV4(policyBase64 string, t time.Time, secretAccessKey, l
 	return signature
 }
 
-// signV4 sign the request before Do(), in accordance with
+// SignV4 sign the request before Do(), in accordance with
 // http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html.
 func SignV4(req http.Request, accessKeyID, secretAccessKey, location string) *http.Request {
 	// Signature calculation is not needed for anonymous credentials.

--- a/signv4/signv4-utils.go
+++ b/signv4/signv4-utils.go
@@ -35,7 +35,7 @@ func xmlDecoder(body io.Reader, v interface{}) error {
 }
 
 // sum256 calculate sha256 sum for an input byte array.
-func Sum256(data []byte) []byte {
+func sum256(data []byte) []byte {
 	hash := sha256.New()
 	hash.Write(data)
 	return hash.Sum(nil)

